### PR TITLE
[INT-9203] ingest user endpoints by label

### PR DESCRIPTION
**Pull request by ryan-willis**
## Summary

Adds a config entry to select hosts by label that will create a `user_endpoint`. If the list is empty, everything is a `Device`:`user_endpoint`. Any hosts that don't match a label will be considered a `Host,Device`:`fleetdm_host`.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by jzolo22 — DISMISSED** [review]

File: src/config.ts [hunk 1]
```diff
@@ -57,6 +63,23 @@ export function parseConfig(
...
+      config.fleetdm_user_endpoint_labels = JSON.parse(origLabels);
+    } else {
+      config.fleetdm_user_endpoint_labels = origLabels.split(',');
+    }
+  } else if (!Array.isArray(origLabels)) {
+    throw new IntegrationValidationError(
+      'Config fleetdm_user_endpoint_labels must be an array of strings',
+    );
+  }
+
```
Comment (mknoedel) [inline src/config.ts:82]: Might be nice to add a log here stating how this property was parsed. I could see how someone might make csv with spaces in it, something like `hello, there` and then ` there` wouldn't work properly.

File: .env.example [hunk 2]
```diff
@@ -1,3 +1,4 @@
 FLEETDM_USER_EMAIL=
 FLEETDM_USER_PASSWORD=
 FLEETDM_HOSTNAME=
+FLEETDM_USER_ENDPOINT_LABELS='[]' # if empty, everything is a user_endpoint
```
Comment (mknoedel) [inline .env.example:4]: This is a reasonably difficult to understand part of this integration, so I'd like to see some documentation in the `docs` section explaining what this is trying to do and the reasoning behind it if possible.

File: docs/jupiterone.md [hunk 3]
```diff
@@ -17,26 +17,31 @@ https://github.com/JupiterOne/sdk/blob/main/docs/integrations/development.md
...
-| Resources | Entity `_type`             | Entity `_class` |
-| --------- | -------------------------- | --------------- |
-| Host      | `fleetdm_hostagent`        | `HostAgent`     |
-| Instance  | `fleetdm_instance`         | `Account`       |
-| Policy    | `fleetdm_policy`           | `ControlPolicy` |
-| Software  | `fleetdm_host_application` | `Application`   |
-| User      | `fleetdm_user`             | `User`          |
+| Resources | Entity `_type`        | Entity `_class`  |
+| --------- | --------------------- | ---------------- |
+| Host      | `fleetdm_host`        | `Host`, `Device` |
```
Comment (mknoedel) [inline docs/jupiterone.md:22]: I'm not sure as I don't really know how fleet dm works, but I suspect that the `fleetdm_host` actually be just be a `Host` and not a `Device` at all. Per the [data model ](https://github.com/JupiterOne/data-model/blob/main/src/schemas/Device.json#L4C26-L4C26), a device is 
> A physical device or media, such as a server, laptop, workstation, smartphone, tablet, router, firewall, switch, wifi-access-point, usb-drive, etc. The exact data type is described in the _type property of the Entity.
and a [host](https://github.com/JupiterOne/data-model/blob/main/src/schemas/Host.json#L4C47-L4C47) is:
> A compute instance that itself owns a whole network stack and serves as an environment for workloads. Typically it runs an operating system. The exact host type is described in the _type property of the Entity. The UUID of the host should be captured in the _id property of the Entity

The lines are super blurry here so I'm more just posing the question. Thoughts?

**Review by mknoedel — DISMISSED** [review]

File: docs/jupiterone.md (hunk 3 above)
Comment (ryan-willis) [reply docs/jupiterone.md:22]: I'm on board with the `fleetdm_host` only being a `Host`

**Review by jzolo22 — APPROVED** [review]

File: src/config.ts (hunk 1 above)
Comment (ryan-willis) [reply src/config.ts]: Added some info logs and updated it + tests to trim

**Pull request merged**